lib/puppet/provider/exec/pwsh.rb

```diff
@@ -26,6 +15,8 @@
   EOT
 
   def run(command, check = false)
+    pwsh = get_pwsh_command
```

Comment:
Are you saying the `:path` attribute cannot take effect when resolving this `commands` block? I thought there were ways to defer execution here using a lambda, but will have to double check that for sure.

Reply:
TLDR Yes.  When `commands` is being evaluated it had no access to the resource so there was no way to get the path.